crates/nu-lsp/src/semantic_tokens.rs

```diff
@@ -0,0 +1,167 @@
+use std::sync::Arc;
+
+use lsp_textdocument::FullTextDocument;
+use lsp_types::{SemanticToken, SemanticTokens, SemanticTokensParams};
+use nu_protocol::{
+    ast::{Block, Expr, Expression, Traverse},
+    engine::StateWorkingSet,
+    Span,
+};
+
+use crate::{span_to_range, LanguageServer};
+
+/// Important for keep spans in increasing order,
+/// since `SemanticToken`s are created by relative positions
+/// to one's previous token
+///
+/// Currently supported types:
+/// 1. internal command names with space
+fn extract_semantic_tokens_from_expression(
+    expr: &Expression,
+    working_set: &StateWorkingSet,
+) -> Option<Vec<Span>> {
+    let closure = |e| extract_semantic_tokens_from_expression(e, working_set);
+    match &expr.expr {
+        Expr::Call(call) => {
+            let command_name_bytes = working_set.get_span_contents(call.head);
+            let head_span = if command_name_bytes.contains(&b' ')
+                && !command_name_bytes.starts_with(b"export")
+                && !command_name_bytes.starts_with(b"overlay")
```

Comment:
Out of curiosity why do they need to be excluded for the purpose of semantic tokens?

May be worth a brief comment for future visitors to the codebase.

Reply:
Sure, tree-sitter-nu makes them keywords, as highlighted differently in screenshot 2, I feel it somewhat bland if all those colors are the same. I'll add some comments

Reply:
Ah ok makes sense! Thanks